File: backend-spring-boot/src/main/java/com/neurotutor/user_service/dto/ModuleItem.java
package com.neurotutor.user_service.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

public class ModuleItem {
    private String id;
    private String titulo;
    private int ejerciciosCompletados;
    private int ejerciciosTotales;
    private String estado;
    private String temaNombre;
    private String nivelRequerido;// EN_CURSO, BLOQUEADO, COMPLETADO
    private int progressPercentage;
    private boolean theoryCompleted;
    private boolean practiceCompleted;
    private boolean examPassed;

    public ModuleItem(String id, String titulo, int ejerciciosCompletados,
                      int ejerciciosTotales, String estado , String temaNombre, String nivelRequerido) {
        this.id = id;
        this.titulo = titulo;
        this.ejerciciosCompletados = ejerciciosCompletados;
        this.ejerciciosTotales = ejerciciosTotales;
        this.estado = estado;
        this.temaNombre=temaNombre;
        this.nivelRequerido = nivelRequerido;
    }

    public ModuleItem(String id, String titulo, int ejerciciosCompletados,
                      int ejerciciosTotales, String estado, String temaNombre,
                      String nivelRequerido, int progressPercentage,
                      boolean theoryCompleted, boolean practiceCompleted,
                      boolean examPassed) {
        this(id, titulo, ejerciciosCompletados, ejerciciosTotales, estado, temaNombre, nivelRequerido);
        this.progressPercentage = progressPercentage;
        this.theoryCompleted = theoryCompleted;
        this.practiceCompleted = practiceCompleted;
        this.examPassed = examPassed;
    }

    // Getters
    public String getId() { return id; }
    public String getTitulo() { return titulo; }
    public int getEjerciciosCompletados() { return ejerciciosCompletados; }
    public int getEjerciciosTotales() { return ejerciciosTotales; }
    public String getEstado() { return estado; }

    public String getTemaNombre() {
        return temaNombre;
    }

    public void setTemaNombre(String temaNombre) {
        this.temaNombre = temaNombre;
    }
    public String getNivelRequerido() {
        return nivelRequerido;
    }

    public void setNivelRequerido(String nivelRequerido) {
        this.nivelRequerido = nivelRequerido;
    }

    @JsonProperty("progress_percentage")
    public int getProgressPercentage() { return progressPercentage; }

    @JsonProperty("theory_completed")
    public boolean isTheoryCompleted() { return theoryCompleted; }

    @JsonProperty("practice_completed")
    public boolean isPracticeCompleted() { return practiceCompleted; }

    @JsonProperty("exam_passed")
    public boolean isExamPassed() { return examPassed; }
}

File: frontend-mobile/app/src/main/java/com/neurotutor/app/mobile/data/model/learning/ModuleItem.kt
package com.neurotutor.app.mobile.data.model.learning

import com.google.gson.annotations.SerializedName

data class ModuleItem(
    @SerializedName("id")
    val id: String,
    
    @SerializedName("titulo")
    val titulo: String,
    
    @SerializedName(value = "tema_nombre", alternate = ["temaNombre"])
    val temaNombre: String,
    
    @SerializedName(value = "nivel_requerido", alternate = ["nivelRequerido"])
    val nivelRequerido: String,
    
    @SerializedName(value = "ejercicios_completados", alternate = ["ejerciciosCompletados"])
    val ejerciciosCompletados: Int,
    
    @SerializedName(value = "ejercicios_totales", alternate = ["ejerciciosTotales"])
    val ejerciciosTotales: Int,
    
    @SerializedName("estado")
    val estado: ModuleStatus,
    
    @SerializedName("progress_percentage")
    val progressPercentage: Int = 0,
    
    @SerializedName("theory_completed")
    val theoryCompleted: Boolean = false,
    
    @SerializedName("practice_completed")
    val practiceCompleted: Boolean = false,
    
    @SerializedName("exam_passed")
    val examPassed: Boolean = false
)
